### PR TITLE
File browser reshuffle 2.0

### PR DESCRIPTION
**Pull request by Commodore64user**
Original PR here #11781

From the makers of the wildly successful Sleep screen rework #11549 and the yet to be merged status bar makeover #11678, We present to you the 'new kid in town', the new file browser (file cabinet) menu.

There has been a far less aggressive approach here, most of the work pertains the reduction of "view modes" and rewording of settings. Some logic was added for the functioning of the simpler view modes but besides that no changes in functionality were added or removed.

I am conflicted about the use of the word "mosaic" here, as what we get is a standard grid view that has nothing to do with actual mosaics, I would love to hear what you think, should it go, should it stay?

the following screenshots were captured on a non-touch kindle 4, so some options might not appear there. Anyway without any further ado, 

Here it is the menu upon opening:

after going to the `Library view`:

Inside `mosaic and list view properties`:

and finally, inside the `file management` submenu:

> [!IMPORTANT]
> I would like to have the classic view properties greyed out when 'classic mode' NOT selected but the usual `enable function` does not work in that case, any help there would be greatly appreciated.

Last but not least, thanks to @Frenzie for his help, patience and support.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/11792)

### CODE REVIEW

**Comment by hius07** [comment]
Look at it please: you can move
https://github.com/koreader/koreader/blob/f894330f1eca81097f9f4c5a5d5c8fd5fb945dea/plugins/coverbrowser.koplugin/main.lua#L296
to the top of `addToMainMenu`, then you will have `fc.display_mode_type` that is false for classic mode, or "list" or "mosaic".
You can use it to shorten all checks.

**Comment by Commodore64user** [comment]
> Look at it please: you can move
> 
> https://github.com/koreader/koreader/blob/f894330f1eca81097f9f4c5a5d5c8fd5fb945dea/plugins/coverbrowser.koplugin/main.lua#L296
> 
> 
> to the top of `addToMainMenu`, then you will have `fc.display_mode_type` that is false for classic mode, or "list" or "mosaic".
> You can use it to shorten all checks.

looking into it

**Comment by ryanwwest** [comment]
Looks cool. 'Mosaic' has always confused me - I'd be open to a name that reminds you more or a grid instead. In general I think one of the biggest hindrances to new users is settings overload, and while I don't want to remove anything, it's always good to make it easier for users to understand the basics.

**Comment by hius07** [comment]
First impressions on wordings and placement:
(1) I am not sure that our file browser is a "library". (Collections are closer to "library" or "bookshelf")
"Items" are changed to "books", but they can be (1) folders, (2) books (supported files), (3) files (unsupported).
(2) I'm fine with Classic - Grid - List.
(3) Do not like "File Management" wording.
(4) Do not like additional "tab" in History tab, Collections tab.
(5) "Mix folders and files" has close relation to sorting, I expect them all together.
(6) Maybe place "Collections and history view" to the end of submenu.
(7) Is "properties" better than "settings"?
(8) We have "mosaic view" and "portrait-mosaic mode".

File: plugins/coverbrowser.koplugin/main.lua [hunk 1]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
     end
     sub_item_table[#modes].separator = true
     table.insert(sub_item_table, {
-        text = _("Use this mode everywhere"),
+        text = _("Hide book covers"),
+        enabled_func = function()
+            return not fc.display_mode_type == false
```
Comment (hius07) [inline plugins/coverbrowser.koplugin/main.lua:137]: We usually write
```Lua
return fc.display_mode_type and true or false
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 2]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                self:setupFileManagerDisplayMode("list_image_filename")
+            end
         end,
-        sub_item_table = collection_sub_item_table,
+        separator = true,
     })
+    ------------------------------------------------------------
+    -- add these settings to File browser Settings submenu
+    if menu_items.filebrowser_settings == nil then return end
+    table.insert(menu_items.filebrowser_settings.sub_item_table, 9, {
```
Comment (hius07) [inline plugins/coverbrowser.koplugin/main.lua:174]: It's the last item, insertion index not needed.

**Comment by hius07** [comment]
We will not be able to have, say, Collections in Mosaic with covers and History in Mosaic without covers, right?

**Comment by poire-z** [comment]
(Haven't yet "read" the new wordings.)

@hius07 and me mentionned around https://github.com/koreader/koreader/discussions/11605#discussioncomment-9135035 that we didn't like not having the 6 modes directly available.
Playing with this PR, I find the "3 modes + toggles" not that bad (but not that better :)).

But, with this change, there are issues:
- `Hide book covers` and `Replace book title with file name` do not stick : when you change mode, they are reset
- This is caused by the fact that you keep using the 6 "modes" values for a single setting in the code - so everytime you change one of the 3 mode/hidecover/usefilename, you  need to generate one of the 6 mode values - and you don't store the state of these 3 mode/hidecover/usefilenameanywhere to be able to reuse them.
- ^ This is what I dislike the most about this PR: the added indirection and gymnastic to build the correct mode in the various menu items callbacks. Before, we had a straight mapping menu item => mode, and it made everything simple and clear. With this PR, we have that added noise and (minor) complexity.
- Proper fix would be to "migrate" this single mode value to a set of 3 view/hide/use settings - but I'd rather not have us involved in implementing this migration. So, it may need another good idea.
- Related, causing a decrease in features: one can't set a different hidecover/usefilename for Filebrowser, History and Collections. I guess not many people would use a different hidecover/usefilename for File browser and History - but I guess one could use filename in filebrowser (a real file browser), while wanting to see metadata in history. So, may be if going on with this and the "sticking" is fixed, the `History tab>` and `Collections tab>` (I'm also not a fan of "tab") submenu should also get their own `[ ] Hide book covers` and `[ ] Replace book title with file name`.
- (^ I hope you just didn't see that, because if you did, and thought _I will just not mention that, if they don't notice it, no one cares, all good_, this is not friendly: better mention every change and risk now if you had seen/think them, so we can discuss and find solutions, rather than get surprises later.)

File: plugins/coverbrowser.koplugin/main.lua [hunk 3]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                enabled_func = function()
+                    return not BookInfoManager:getSetting("unified_display_mode")
+                end,
+                sub_item_table = collection_sub_item_table,
+                separator = true,
+            }
+        }
+    })
+
+    table.insert(sub_item_table, menu_items.filebrowser_settings.sub_item_table[5])
```
Comment (hius07) [inline plugins/coverbrowser.koplugin/main.lua:288]: A comment "Classic view properties" would be nice.

File: plugins/coverbrowser.koplugin/main.lua [hunk 4]
```diff
@@ -330,36 +478,52 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                        text = _("Show visual cue for books with descriptions"),
                         checked_func = function() return not BookInfoManager:getSetting("no_hint_description") end,
                         callback = function()
                             BookInfoManager:toggleSetting("no_hint_description")
                             fc:updateItems(1, true)
                         end,
                     },
                     {
-                        text = _("Show hint for book status in history"),
+                        text = _("Show visual cue for books in history tab"),
```
Comment (hius07) [inline plugins/coverbrowser.koplugin/main.lua:518]: This is incorrect text change.

File: plugins/coverbrowser.koplugin/main.lua [hunk 5]
```diff
@@ -330,36 +478,52 @@ function CoverBrowser:addToMainMenu(menu_items)
...
                         checked_func = function() return BookInfoManager:getSetting("history_hint_opened") end,
                         callback = function()
                             BookInfoManager:toggleSetting("history_hint_opened")
                             fc:updateItems(1, true)
                         end,
                     },
                     {
-                        text = _("Show hint for book status in collections"),
+                        text = _("Show visual cue for books added to collections"),
```
Comment (hius07) [inline plugins/coverbrowser.koplugin/main.lua:526]: This is incorrect text change.

File: plugins/coverbrowser.koplugin/main.lua [hunk 6]
```diff
@@ -330,36 +478,52 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                            return not BookInfoManager:getSetting("hide_page_info") and fc.display_mode_type == "list"
+                        end,
                         checked_func = function() return BookInfoManager:getSetting("show_pages_left_in_progress") end,
                         callback = function()
                             BookInfoManager:toggleSetting("show_pages_left_in_progress")
                             fc:updateItems(1, true)
                         end,
                     },
+                    {
+                        text = _("Show file metadata"),
```
Comment (hius07) [inline plugins/coverbrowser.koplugin/main.lua:491]: I don't think file extension and size are "metadata".

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 7]
```diff
@@ -149,7 +149,7 @@ function FileManagerMenu:setUpdateItemTable()
 
     -- setting tab
     self.menu_items.filebrowser_settings = {
-        text = _("Settings"),
+        text = _("File Management"),
```
Comment (Frenzie) [inline frontend/apps/filemanager/filemanagermenu.lua:152]: Settings is much clearer. Settings is something you ignore unless you're looking for settings. File management implies a great many things, all of which are incorrect.

File: plugins/coverbrowser.koplugin/main.lua (hunk 6 above)
Comment (Frenzie) [reply plugins/coverbrowser.koplugin/main.lua:491]: Agreed. These are file properties, not metadata. (You might argue that properties are a specific subset of metadata in which case… yes, that's the point. :-)

**Comment by Commodore64user** [comment]
> * `Hide book covers` and `Replace book title with file name` do not stick : when you change mode, they are reset
> * This is caused by the fact that you keep using the 6 "modes" values for a single setting in the code - so everytime you change one of the 3 mode/hidecover/usefilename, you  need to generate one of the 6 mode values - and you don't store the state of these 3 mode/hidecover/usefilenameanywhere to be able to reuse them.
> * ^ This is what I dislike the most about this PR: the added indirection and gymnastic to build the correct mode in the various menu items callbacks. Before, we had a straight mapping menu item => mode, and it made everything simple and clear. With this PR, we have that added noise and (minor) complexity.

you are not wrong there, I know this could have been written in a much nicer way, but I didn't want to be asking for help as I know another approach would be much more difficult given my level of experience (or lack of really)

> * Proper fix would be to "migrate" this single mode value to a set of 3 view/hide/use settings - but I'd rather not have us involved in implementing this migration. So, it may need another good idea.

wink, exactly what I was saying before.

> * Related, causing a decrease in features: one can't set a different hidecover/usefilename for Filebrowser, History and Collections. I guess not many people would use a different hidecover/usefilename for File browser and History - but I guess one could use filename in filebrowser (a real file browser), while wanting to see metadata in history. So, may be if going on with this and the "sticking" is fixed, the `History tab>` and `Collections tab>` (I'm also not a fan of "tab") submenu should also get their own `[ ] Hide book covers` and `[ ] Replace book title with file name`.

what do you mean? you can certainly still do that, well with the caveat of the `[ ] Hide book covers` and `[ ] Replace book title with file name` pointed out. Which leads to me to the last point,

> * (^ I hope you just didn't see that, because if you did, and thought _I will just not mention that, if they don't notice it, no one cares, all good_, this is not friendly: better mention every change and risk now if you had seen/think them, so we can discuss and find solutions, rather than get surprises later.)

That is quite the accusation there @poire-z, slow down. I am not trying to hide anything from anyone, I think I have been open about issues in the past https://github.com/koreader/koreader/pull/11586#issuecomment-2016836836. that just went over the radar.

File: frontend/apps/filemanager/filemanagermenu.lua (hunk 7 above)
Comment (Commodore64user) [reply frontend/apps/filemanager/filemanagermenu.lua:152]: my issue lies with many places using "Settings" (not like "x settings") but literally just "settings", I would like to reduce it to just one. I am happy to keep it as "settings" but the "settings" elsewhere would eventually go away.

File: plugins/coverbrowser.koplugin/main.lua (hunk 2 above)
Comment (Commodore64user) [reply plugins/coverbrowser.koplugin/main.lua:174]: good point.

File: plugins/coverbrowser.koplugin/main.lua (hunk 3 above)
Comment (Commodore64user) [reply plugins/coverbrowser.koplugin/main.lua:288]: will do

File: plugins/coverbrowser.koplugin/main.lua (hunk 4 above)
Comment (Commodore64user) [reply plugins/coverbrowser.koplugin/main.lua:518]: I think I finally figure out what these two (this and the next) options do.

I always assumed the dog ear meant book open. I didn't realised it had different meanings across the ui, that is probably not a good thing.

will change the texts though

File: plugins/coverbrowser.koplugin/main.lua (hunk 5 above)
Comment (Commodore64user) [reply plugins/coverbrowser.koplugin/main.lua:526]: ditto

File: plugins/coverbrowser.koplugin/main.lua (hunk 1 above)
Comment (Commodore64user) [reply plugins/coverbrowser.koplugin/main.lua:137]: what about `return fc.display_mode_type == false` ?

Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:137]:
```suggestion
            return fc.display_mode_type and true or false
```

File: frontend/apps/filemanager/filemanagermenu.lua (hunk 7 above)
Comment (Commodore64user) [inline frontend/apps/filemanager/filemanagermenu.lua:152]:
```suggestion
        text = _("Settings"),
```

File: plugins/coverbrowser.koplugin/main.lua (hunk 2 above)
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:174]:
```suggestion
    table.insert(menu_items.filebrowser_settings.sub_item_table, {
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 8]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                text = _("Collections tab"),
+                enabled_func = function()
+                    return not BookInfoManager:getSetting("unified_display_mode")
+                end,
+                sub_item_table = collection_sub_item_table,
+                separator = true,
+            }
+        }
+    })
+
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:287]:
```suggestion

-- Classic view properties
```

File: plugins/coverbrowser.koplugin/main.lua (hunk 6 above)
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:491]:
```suggestion
                        text = _("Show file properties"),
```

Comment (Commodore64user) [reply plugins/coverbrowser.koplugin/main.lua:491]: I wasn't going to argue that, happy to change it, in fact I thought I had changed it back to the original as at some point had used a way worse one, alas I didn't.

File: plugins/coverbrowser.koplugin/main.lua (hunk 5 above)
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:526]:
```suggestion
                        text = _("Show visual cue for open books in collections"),
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 9]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+    table.insert(sub_item_table, menu_items.filebrowser_settings.sub_item_table[5])
+    table.remove(menu_items.filebrowser_settings.sub_item_table, 5)
+
+    table.insert(sub_item_table, {
+        text = _("Mosaic and list view properties"),
         separator = true,
+        enabled_func = function()
+            return not fc.display_mode_type == false
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:295]:
```suggestion
            return fc.display_mode_type and true or false
```

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 10]
```diff
@@ -272,7 +290,7 @@ function FileManagerMenu:setUpdateItemTable()
                 },
             },
             {
-                text = _("History settings"),
+                text = _("History tab"),
```
Comment (Commodore64user) [inline frontend/apps/filemanager/filemanagermenu.lua:293]:
```suggestion
                text = _("History"),
```

File: plugins/coverbrowser.koplugin/main.lua (hunk 4 above)
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:518]:
```suggestion
                        text = _("Show visual cue for open books in history"),
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 11]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                end,
+                callback = function()
+                    if BookInfoManager:toggleSetting("unified_display_mode") then
+                        self:setupHistoryDisplayMode(filemanager_display_mode)
+                        self:setupCollectionDisplayMode(filemanager_display_mode)
+                    end
+                end,
+            },
+            {
+                text = _("History tab"),
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:271]:
```suggestion
                text = _("History"),
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 12]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                enabled_func = function()
+                    return not BookInfoManager:getSetting("unified_display_mode")
+                end,
+                sub_item_table = history_sub_item_table,
+            },
+            {
+                text = _("Collections tab"),
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:278]:
```suggestion
                text = _("Collections"),
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 13]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                separator = true,
+            }
+        }
+    })
+
+    table.insert(sub_item_table, menu_items.filebrowser_settings.sub_item_table[5])
+    table.remove(menu_items.filebrowser_settings.sub_item_table, 5)
+
+    table.insert(sub_item_table, {
+        text = _("Mosaic and list view properties"),
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:292]:
```suggestion
        text = _("Grid and list view properties"),
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 14]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
         separator = true,
+        enabled_func = function()
+            return not fc.display_mode_type == false
+        end,
         sub_item_table = {
             {
                 text_func = function()
-                    return T(_("Items per page in portrait mosaic mode: %1 × %2"), fc.nb_cols_portrait, fc.nb_rows_portrait)
+                    return T(_("Grid size in portrait-mosaic mode: %1 × %2"), fc.nb_cols_portrait, fc.nb_rows_portrait)
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:300]:
```suggestion
                    return T(_("Grid size in portrait mode: %1 × %2"), fc.nb_cols_portrait, fc.nb_rows_portrait)
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 15]
```diff
@@ -215,14 +349,17 @@ function CoverBrowser:addToMainMenu(menu_items)
             },
             {
                 text_func = function()
-                    return T(_("Items per page in landscape mosaic mode: %1 × %2"), fc.nb_cols_landscape, fc.nb_rows_landscape)
+                    return T(_("Grid size in landscape-mosaic mode: %1 × %2"), fc.nb_cols_landscape, fc.nb_rows_landscape)
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:352]:
```suggestion
                    return T(_("Grid size in landscape mode: %1 × %2"), fc.nb_cols_landscape, fc.nb_rows_landscape)
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 16]
```diff
@@ -126,55 +132,183 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                    return fc.display_mode_type == "mosaic"
                 end,
                 -- Best to not "keep_menu_open = true", to see how this apply on the full view
                 callback = function()
                     local nb_cols = fc.nb_cols_portrait
                     local nb_rows = fc.nb_rows_portrait
                     local DoubleSpinWidget = require("/ui/widget/doublespinwidget")
                     local widget = DoubleSpinWidget:new{
-                        title_text = _("Portrait mosaic mode"),
+                        title_text = _("Portrait mosaic view"),
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:311]:
```suggestion
                        title_text = _("Portrait grid view"),
```

File: plugins/coverbrowser.koplugin/main.lua [hunk 17]
```diff
@@ -215,14 +349,17 @@ function CoverBrowser:addToMainMenu(menu_items)
...
+                enabled_func = function()
+                    return fc.display_mode_type == "mosaic"
                 end,
                 callback = function()
                     local nb_cols = fc.nb_cols_landscape
                     local nb_rows = fc.nb_rows_landscape
                     local DoubleSpinWidget = require("/ui/widget/doublespinwidget")
                     local widget = DoubleSpinWidget:new{
-                        title_text = _("Landscape mosaic mode"),
+                        title_text = _("Landscape mosaic view"),
```
Comment (Commodore64user) [inline plugins/coverbrowser.koplugin/main.lua:362]:
```suggestion
                        title_text = _("Landscape grid view"),
```

File: plugins/coverbrowser.koplugin/main.lua (hunk 1 above)
Comment (hius07) [reply plugins/coverbrowser.koplugin/main.lua]: That's opposite to what you need.

Comment (Commodore64user) [reply plugins/coverbrowser.koplugin/main.lua]: Yes i know, what i am asking is: would you write that in a different way too?

Comment (hius07) [reply plugins/coverbrowser.koplugin/main.lua]: For this case it can be
`return fc.display_mode_type ~= nil`

**Comment by Frenzie** [comment]
> again, any other ideas? I was actually quite proud of that one.

The obvious alternative to just `settings` is `file browser settings`.

**Comment by Commodore64user** [comment]
> > again, any other ideas? I was actually quite proud of that one.
> 
> 
> 
> The obvious alternative to just `settings` is `file browser settings`.

I don't mind "Settings" being used here, so long as all the other ones slowly disappear -one by one-. Actually it makes some sense to use it here where it is broad and general.

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 18]
```diff
@@ -165,26 +165,41 @@ function FileManagerMenu:setUpdateItemTable()
...
+                    return collate.can_collate_mixed and G_reader_settings:isTrue("collate_mixed")
+                end,
+                callback = function()
+                    G_reader_settings:flipNilOrFalse("collate_mixed")
+                    FileChooser:refreshPath()
+                end,
             },
             {
-                text = _("Classic mode settings"),
+                text = _("Classic view properties"),
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:186]: Proposed change: "Classic view settings".

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 19]
```diff
@@ -165,26 +165,41 @@ function FileManagerMenu:setUpdateItemTable()
...
                 sub_item_table = {
                     {
                         text_func = function()
-                            return T(_("Items per page: %1"),
+                            return T(_("Book entries per page: %1"),
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:190]: Proposed change: "Entries per page".

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 20]
```diff
@@ -201,14 +216,17 @@ function FileManagerMenu:setUpdateItemTable()
                     },
                     {
                         text_func = function()
-                            return T(_("Item font size: %1"), FileChooser.font_size)
+                            local items_per_page = G_reader_settings:readSetting("items_per_page") or FileChooser.items_per_page_default
+                            local items_font_size = FileChooser.font_size
+                                or G_reader_settings:readSetting("items_font_size") or FileChooser.getItemFontSize(items_per_page)
+                            return T(_("File name font size: %1"), items_font_size)
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:222]: Proposed change: "Entry font size".

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 21]
```diff
@@ -201,14 +216,17 @@ function FileManagerMenu:setUpdateItemTable()
...
                         callback = function(touchmenu_instance)
-                            local current_value = FileChooser.font_size
-                            local default_value = FileChooser.getItemFontSize(G_reader_settings:readSetting("items_per_page")
-                                or FileChooser.items_per_page_default)
+                            local items_per_page = G_reader_settings:readSetting("items_per_page") or FileChooser.items_per_page_default
+                            local default_value = FileChooser.getItemFontSize(items_per_page)
+                            local current_value = FileChooser.font_size or G_reader_settings:readSetting("items_font_size") or default_value
                             local widget = SpinWidget:new{
-                                title_text =  _("Item font size"),
+                                title_text =  _("File name font size"),
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:229]: Proposed change: "Entry font size".

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 22]
```diff
@@ -231,7 +249,7 @@ function FileManagerMenu:setUpdateItemTable()
                         end,
                     },
                     {
-                        text = _("Shrink item font size to fit more text"),
+                        text = _("Auto reduce filename-font-size to fit whole text"),
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:252]: Proposed change: "Auto reduce font size to fit more text" (we do not guarantee to fit the **whole** text).

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 23]
```diff
@@ -295,10 +313,10 @@ function FileManagerMenu:setUpdateItemTable()
                         separator = true,
                     },
                     {
-                        text = _("Clear history of deleted files"),
+                        text = _("Clear deleted files from history"),
                         callback = function()
                             UIManager:show(ConfirmBox:new{
-                                text = _("Clear history of deleted files?"),
+                                text = _("Clear deleted files from history?"),
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:319]: Proposed change: "Remove deleted files from history".

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 24]
```diff
@@ -307,7 +325,7 @@ function FileManagerMenu:setUpdateItemTable()
                         end,
                     },
                     {
-                        text = _("Auto-remove deleted or purged items from history"),
+                        text = _("Auto-remove deleted or purged entries from history"),
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:328]: Proposed change: "Auto-remove deleted or reset files from history".

**Comment by hius07** [comment]
CircleCI warnings:
```
plugins/coverbrowser.koplugin/main.lua:96:66: line contains trailing whitespace
plugins/coverbrowser.koplugin/main.lua:102:21: line contains trailing whitespace
plugins/coverbrowser.koplugin/main.lua:151:16: line contains trailing whitespace
plugins/coverbrowser.koplugin/main.lua:471:50: line contains trailing whitespace
```

**Comment by poire-z** [comment]
> * `Hide book covers` and `Replace book title with file name` do not stick : when you change mode, they are reset

I think you could still easily get Hide book covers to stick when switching between Grid & list. But if you switch to Classic, you currently can't know what was the previous state.

> > * Related, causing a decrease in features: one can't set a different hidecover/usefilename for Filebrowser, History and Collections. I guess not many people would use a different hidecover/usefilename for File browser and History - but I guess one could use filename in filebrowser (a real file browser), while wanting to see metadata in history. So, may be if going on with this and the "sticking" is fixed, the `History tab>` and `Collections tab>` (I'm also not a fan of "tab") submenu should also get their own `[ ] Hide book covers` and `[ ] Replace book title with file name`.
> 
> what do you mean? you can certainly still do that, well with the caveat of the `[ ] Hide book covers` and `[ ] Replace book title with file name` pointed out. Which leads to me to the last point,

I mean just that: the caveat of the `[ ] Hide book covers` and `[ ] Replace book title with file name`.

Have you got an idea on how to solve that?
You can't keep using `if filemanager_display_mode` when handling stuff for History and Collections. Or one would need to go back to filemanager, set Hide covers there, go back 2 submenu down to History to select Grid to get Grid text book cover, then go back to reset it to how he wants it for FileManager. That's too twisted and uglier than the current simple way to handle it.

Your changes are too filebrowser centric. If you want stuff to be intuitive for History and Collections, you may need to triplicate lots of stuff there.

Also, you can't use enabled_func on the `Grid size in portrait mosaic mode>` depending on only if mosaic is enabled or not for FileBrowser, because then, if you have Grid mode enabled for History, you can't tweak the grid size anymore.
So these settings would need to be always enabled, or triplicated too.

Even if I don't dislike the intended look & feel, I don't think it's worth all these _complications_ :)

**Comment by Commodore64user** [comment]
okay, back here. I have been thinking about this, how to solve the issues and so forth.

> I think you could still easily get Hide book covers to stick when switching between Grid & list. But if you switch to Classic, you currently can't know what was the previous state.

I was considering saving to G_settings and reading it.

> You can't keep using if filemanager_display_mode when handling stuff for History and Collections. Or one would need to go back to filemanager, set Hide covers there, go back 2 submenu down to History to select Grid to get Grid text book cover, then go back to reset it to how he wants it for FileManager. That's too twisted and uglier than the current simple way to handle it.

I hear you, well read you.

> Your changes are too filebrowser centric. If you want stuff to be intuitive for History and Collections, you may need to triplicate lots of stuff there.

lets not.

> Also, you can't use enabled_func on the Grid size in portrait mosaic mode> depending on only if mosaic is enabled or not for FileBrowser, because then, if you have Grid mode enabled for History, you can't tweak the grid size anymore.
So these settings would need to be always enabled, or triplicated too.

I hadn't consider this either. that steams mainly from alway using the same view for history and ~~favourites~~ collections. and if I am honest, I also never go to history or use collections so...

> Even if I don't dislike the intended look & feel, I don't think it's worth all these complications :)
> * Proper fix would be to "migrate" this single mode value to a set of 3 view/hide/use settings - but I'd rather not have us involved in implementing this migration. So, it may need another good idea.

well it would certainly be easier if you could help. I don't quite get your reticence (see below) so slow steps.

> * Proper fix would be to "migrate" this single mode value to a set of 3 view/hide/use settings - but I'd rather not have us involved in implementing this migration. So, it may need another good idea.

**Comment by hius07** [comment]
> another good idea

Keep the menu structure as is, change the wordings only.

**Comment by poire-z** [comment]
> I was considering saving to G_settings and reading it.

That would be an ugly option.
For historical reasons, this plugin saves its settings in the db - at the time, I was strongly suggested to not push crap in G_reader_settings, so I stored them in the plugin db, as it happened it has one. Since then, lots of plugins just don't care and use G_reader_settings, which is fine by me.
But if the plugin should now be allowed to use G_reader_settings, all of its settings should go there - not only little helper settings like the one you need. So, it's another bigger migration. Another lingering idea is to make this plugin part of core, which would allow its metadata db to be of more general use. And that's something bigger, that we can't let to you, and that would require @hius07 or me to have the time and energy to focus on this huge task.
I'm quite against doing "slow little steps", when we have no real time/interest/need/envy/urge to focus on the target situation to be sure these little steps will be the right ones.

> well it would certainly be easier if you could help. I don't quite get your reticence (see below) so slow steps.

Above is my reticence: doing things right would need more conviction and interest and work and focus (and time, that I can't afford at the moment), and copiloting you would be painful, for both of us. (I'd rather waste myself a weekend and do it alone myself).
Moreover, even if I don't dislike your menu layout, your enthusiasm does not look like it is really shared :) and migrating settings to a set that fit your model would feel a bit arbitrary. Tomorrow, we could get the visit of AmstradCpcUser that  will suggest another menu layout, and there comes a new migration of settings to fit better the new model...
To be honest, your model is quite right: we have 3 individual modules that do classic/grid/list, everything else are parameters to them. But still a lot.

Anyway, some ideas that would not require all the above suggested hard work (I hope I've been scary enough:)), keeping this as a UI only job:

You could store the individual state of hide_cover/use_filename in an internal variable, no need to save it, it would be remembered only for the current menu session. When the menu is first opened and these vars are not there, you would compute their initial values from the initial mode. I think it's fine if we don't remember that when one month ago the use tested Grid with hide covers, and settled on List with covers, if he now switches to Grid he gets covers. You just need that if now when testing stuff, when he has hideCover, when he switches from Grid to List to Classic to Grid, he gets a sticky hideCover.

If you don't want to triplicate things, the big loop that builds the 3 menu items (filebrowser, history, collections) should be reworked a bit, and should also build the Hide cover/Filename only menu items in the 3 submenus - setting the right initial values to the mentionned settings from the proper fb/history/collection current mode.

**Comment by Frenzie** [comment]
> at the time, I was strongly suggested to not push crap in G_reader_settings

Sorry (probably). :-)

**Comment by Commodore64user** [comment]
> You could store the individual state of hide_cover/use_filename in an internal variable, no need to save it, it would be remembered only for the current menu session. When the menu is first opened and these vars are not there, you would compute their initial values from the initial mode. I think it's fine if we don't remember that when one month ago the use tested Grid with hide covers, and settled on List with covers, if he now switches to Grid he gets covers. You just need that if now when testing stuff, when he has hideCover, when he switches from Grid to List to Classic to Grid, he gets a sticky hideCover.
>
> If you don't want to triplicate things, the big loop that builds the 3 menu items (filebrowser, history, collections) should be reworked a bit, and should also build the Hide cover/Filename only menu items in the 3 submenus - setting the right initial values to the mentioned settings from the proper fb/history/collection current mode.

So I tried to make them "stick" and to adjust the for loops and _failed miserably_, so two options, three really.

1. somebody else takes the helm and fixes it (ideal)
2. change back to the ugly 6 modes (same as before but with the wording changes elsewhere)
3. close this and forget about the whole thing (demolish the house, with the cat and dog inside)

**Pull request closed without merging**